### PR TITLE
Variant and and variant attribute support

### PR DESCRIPTION
**Pull request by sanmut**
Add support for variant datatype and primitives DataToVariant and
VariantToData.
Add Support Get/Set/GetAll/Delete Variant Attributes

### CODE REVIEW

File: source/core/Variants.cpp [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+
+        if (variantType->IsA(destType)) {
+            void* destData = _ParamImmediate(DestData)._pData;
+            variantType->CopyData(variantType->Begin(kPARead), destData);
+        } else {
+            if (errPtr) {
+                errPtr->SetError(true, 91, "Variant To Data");
+            }
+        }
+    }
```
Comment (spathiwa) [inline source/core/Variants.cpp]: The primitive write default data to its output when an error is passed in, or when the destType doesn't match the variant's contained type.

File: source/core/Variants.cpp [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+    if (_ParamPointer(Replaced)) {
+        _Param(Replaced) = replaced;
+    }
+    return _NextInstruction();
+}
+
+struct GetVariantAttributesParamBlock : public VarArgInstruction
```
Comment (spathiwa) [inline source/core/Variants.cpp]: Name should be singular, returns only one attribute.

File: source/core/Variants.cpp [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+    TypeRef variantType = DefaultValueType::New(tm, inputType, true);
+    variantType->CopyData(_ParamImmediate(InputData._pData), variantType->Begin(kPAWrite));
+    _Param(OutputVariant) = variantType;
+    return _NextInstruction();
+}
+
+//------------------------------------------------------------
+struct VariantToDataParamBlock : public VarArgInstruction
+{
+    _ParamDef(TypeRef, VariantType);
```
Comment (spathiwa) [inline source/core/Variants.cpp]: Rename to just Variant?  It provides an input value, not just a type.

File: test-it/ViaTests/VariantCopy.via [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        PrintVariantWithAttributes(myVar2 "myVar2")
+
+        DeleteVariantAttribute(myVar * found error)
+        Printf("After Delete in myVar():\n")
+        PrintVariantWithAttributes(myVar "myVar")
+        PrintVariantWithAttributes(myVar2 "myVar2")
+    )
+)))
+
+enqueue(TestCopyVariantWithAttributes)
```
Comment (cglzaguilar) [inline test-it/ViaTests/VariantCopy.via]: Add a new line to remove the "No newline at end of file" red glyph on git review window.

File: source/core/Variants.cpp (hunk 2 above)
Comment (sanmut) [reply source/core/Variants.cpp]: Fixed. Thanks.

**Pull request merged**